step-api/src/main/java/org/jenkinsci/plugins/workflow/steps/AbstractSynchronousNonBlockingStepExecution.java

```diff
@@ -61,21 +78,4 @@ private static synchronized ExecutorService getExecutorService() {
         return executorService;
     }
 
-    private static class StepRunner implements Runnable {
```

Comment:
Was it needed? I see it clearer by having the run logic inside this private inner class. It would suffice to pass the `auth` object in the constructor and store it in a local field.